numpy/core/src/multiarray/mapping.h

```diff
@@ -3,6 +3,15 @@
 
 extern NPY_NO_EXPORT PyMappingMethods array_as_mapping;
 
+/*
+ * Plain indexing is python code writing arr[...]. Fancy means explicitly
+ * the old behavious, legacy is mostly like fancy, but does not accept
+ * array-likes as tuples.
+ */
+#define PLAIN_INDEXING 1
+#define OUTER_INDEXING 2
+#define VECTOR_INDEXING 4
+#define FANCY_INDEXING 8
```

Comment:
This seems to imply that these can be or'd together - does that make any sense at all?

Reply:
Yeah, you are right, but before going too much in depth, might want to mostly discuss the gist. There are also larger code blocks that need to be deleted due to not support bools in vindex. Its pretty alpha, though with code removal, and disabling that `prepared` thingy, it should probably work fine. (plus making the warnings more conservative).